### PR TITLE
GLT-1021: description field modifications

### PR DESCRIPTION
**Pull request by djcooke**
* Removed description field from
  * SamplePurpose
  * TissueMaterial
* Made description nullable for
  * Sample
  * Library
  * Pool
  * Run
  * Experiment
  * Platform
  * Subproject
  * Study
* Made Run alias NOT nullable

### CODE REVIEW

File: miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java [hunk 1]
```diff
@@ -381,7 +381,6 @@ private DetailedSample findOrCreateParent(DetailedSample sample) throws IOExcept
   private SampleTissue createParentTissue(SampleTissue tissue, DetailedSample child) throws IOException {
     log.debug("Creating a new Tissue to use as a parent.");
     tissue.setProject(child.getProject());
-    tissue.setDescription("Tissue");
```
Comment (callunity) [inline miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java]: Might it be worth keeping a description in here to indicate that these are ghost tissues/stocks?

Comment (djcooke) [inline miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java]: That's what the 'isSynthetic' field is for, and a big grey box explaining it shows up in the UI. The ghosts' old descriptions didn't mention that they were ghosts anyway - just the sample category, which again is available elsewhere

**Comment by djcooke** [comment]
I think the difference between null and empty description doesn't really matter. I could remove the statements changing empty->null. There's no changelog entry when changing null->empty, and Pinery will see it the same either way

File: miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java (hunk 1 above)
Comment (djcooke) [inline miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java]: Want to make a separate ticket for making better use of the 'isSynthetic' field? I don't think storing the same information in two places - one of which is freely-editable free text (description) - is the solution

**Comment by callunity** [comment]
Ok, it looks like the issue I'm thinking of got fixed a while ago.

File: miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java (hunk 1 above)
Comment (callunity) [inline miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java]: I will make the ticket. I agree it's not the solution, or even a good solution, but I think adding "ghost" to the description is a user-friendly stopgap until we ultimately make better use of the 'isSynthetic' field.

File: miso-web/src/main/webapp/scripts/pool_ajax.js [hunk 2]
```diff
@@ -52,8 +52,7 @@ var Pool = Pool || {
 
     // Description input field validation
     jQuery('#description').attr('class', 'form-control');
-    jQuery('#description').attr('data-parsley-required', 'true');
-    jQuery('#description').attr('data-parsley-maxlength', '100');
+    jQuery('#description').attr('data-parsley-maxlength', '255');
```
Comment (callunity) [inline miso-web/src/main/webapp/scripts/pool_ajax.js]: I've just noticed that Pool volume is also set to `required`, would you mind changing that in this PR?

**Comment by callunity** [comment]
I extra approve this PR... (oops)

File: miso-web/src/main/webapp/scripts/pool_ajax.js (hunk 2 above)
Comment (djcooke) [inline miso-web/src/main/webapp/scripts/pool_ajax.js]: changed

**Pull request merged**